### PR TITLE
cephtool/test.sh error on full tests

### PR DESCRIPTION
**Pull request by dzafman**
Fix to make cephtool/test.sh run during rados suite.

### CODE REVIEW

File: qa/workunits/ceph-helpers.sh [hunk 1]
```diff
@@ -1297,6 +1297,48 @@ function test_wait_for_clean() {
...
+        fi
+        sleep ${delays[$loop]}
+        loop+=1
+    done    
+    return 0
+}
+
+function wait_for_health_ok() {
+     wait_for_health "HEALTH_OK" || return 1
+     return 0
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: `return 0` is not necessary here.

File: qa/workunits/ceph-helpers.sh [hunk 2]
```diff
@@ -1297,6 +1297,48 @@ function test_wait_for_clean() {
...
+#
+# @return 0 if the cluster is HEALTHY, 1 otherwise
+#
+function wait_for_health() {
+    local grepstr=$1
+    local -a delays=($(get_timeout_delays $TIMEOUT .1))
+    local -i loop=0
+
+    while ! ceph health detail | grep "$grepstr" ; do
+        ceph health detail
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: shall we wait a little bit before retrying "ceph health detail". in other words, is the "ceph health detail" necessary here?

File: qa/workunits/ceph-helpers.sh [hunk 3]
```diff
@@ -1297,6 +1297,48 @@ function test_wait_for_clean() {
...
+	if (( $loop >= ${#delays[*]} )) ; then
+            return 1
+        fi
+        sleep ${delays[$loop]}
+        loop+=1
+    done    
+    return 0
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: if `ceph health detail | grep "$grepstr"` returns 0, we don't need to `return 0` explicitly.

File: qa/workunits/ceph-helpers.sh [hunk 4]
```diff
@@ -1607,6 +1649,19 @@ function test_wait_background() {
...
 }
 
+function get_admin_socket()
+{
+  local client=$1
+
+  if test -n "$CEPH_OUT_DIR";
+  then
+    echo $CEPH_OUT_DIR/$client.asok
+    return 0
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: not necessary.

File: qa/workunits/ceph-helpers.sh [hunk 5]
```diff
@@ -1607,6 +1649,19 @@ function test_wait_background() {
...
+  fi
+  echo "/var/run/ceph/ceph-$client.asok"
+  return 0
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: no need to return 0 here explicitly.

File: qa/workunits/ceph-helpers.sh [hunk 6]
```diff
@@ -1607,6 +1649,19 @@ function test_wait_background() {
...
+function get_admin_socket()
+{
+  local client=$1
+
+  if test -n "$CEPH_OUT_DIR";
+  then
+    echo $CEPH_OUT_DIR/$client.asok
+    return 0
+  fi
+  echo "/var/run/ceph/ceph-$client.asok"
```
Comment (tchaikov) [inline qa/workunits/ceph-helpers.sh]: better off putting this `echo` in the `else` branch.

File: qa/workunits/cephtool/test.sh [hunk 7]
```diff
@@ -1438,36 +1441,32 @@ function test_mon_pg()
...
-  WAITFORFULL=10
-  ceph --admin-daemon $CEPH_OUT_DIR/osd.0.asok injectfull nearfull
-  sleep $WAITFORFULL
-  ceph health | grep "HEALTH_WARN.*1 nearfull osd(s)"
-  ceph --admin-daemon $CEPH_OUT_DIR/osd.1.asok injectfull backfillfull
-  sleep $WAITFORFULL
-  ceph health | grep "HEALTH_WARN.*1 backfillfull osd(s)"
-  ceph --admin-daemon $CEPH_OUT_DIR/osd.2.asok injectfull failsafe
-  sleep $WAITFORFULL
+  ceph --admin-daemon $(get_admin_socket osd.0) injectfull nearfull
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh]: can we just use 
```sh
ceph daemon osd.0 injectfull nearfull
```
and let ceph cli to figure out the asock's path?

File: qa/workunits/cephtool/test.sh [hunk 8]
```diff
@@ -1442,10 +1442,12 @@ function test_mon_pg()
   # Check health status
   ceph osd set-nearfull-ratio .913
   ceph health | grep 'HEALTH_ERR.*Full ratio(s) out of order'
-  ceph health detail | grep 'backfillfull_ratio (0.912) < nearfull_ratio (0.913), increased'
+  ceph health detail
+  ceph health detail | grep 'backfillfull_ratio (0.912) < nearfull_ratio (0.913), increased' || true
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh]: @dzafman is this for debugging purpose?

File: qa/workunits/ceph-helpers.sh [hunk 9]
```diff
@@ -1607,6 +1649,19 @@ function test_wait_background() {
     if [ ! -z "$pids" ]; then return 1; fi
 }
 
+function get_admin_socket()
+{
```
Comment (dachary) [inline qa/workunits/ceph-helpers.sh]: Most ceph-helper functions have a **dir** first argument and work relative to it. It's goot to have a get_admin_socket so the caller does not have to hardwire $dir/$client.asok.

<pre>
function get_admin_socket() {
  local dir=$1
  local client=$1
..
</pre>

File: qa/workunits/ceph-helpers.sh (hunk 2 above)
Comment (dachary) [inline qa/workunits/ceph-helpers.sh]: I agree that it is redundant.

File: qa/workunits/cephtool/test.sh (hunk 7 above)
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh]: > # A **dir**/ceph.conf file is created but not meant to be used by any
> # function.  It is convenient for debugging a failure with:
> #
> #     ceph --conf **dir**/ceph.conf -s

oh, sorry, i was wrong. the `ceph.conf` is "not meant to be used". so scratch that.

File: qa/workunits/ceph-helpers.sh (hunk 9 above)
Comment (dzafman) [inline qa/workunits/ceph-helpers.sh]: The cephtool/test.sh is being run as part of make check as well as in the rados suite.  Previously, nobody was trying to find the admin socket in code like that anywhere I found.  The code needs to check for the $CEPH_* environment variables to determine if we are in make check or rados suite.  The rados suite is the one that uses the $dir everywhere but the other does not.  Suggestions?

Comment (dachary) [inline qa/workunits/ceph-helpers.sh]: all functions in ceph-helpers.sh are consistent with how the run_mon and run_osd setup the directories. If making get_admin_socket comply with that is bizarre in the context of test.sh, maybe the function should be defined at the beginning of test.sh instead (there are a few already in this case).

File: qa/workunits/ceph-helpers.sh (hunk 2 above)
Comment (dzafman) [inline qa/workunits/ceph-helpers.sh]: I'll move this to happen only if we are going to fail so we can see that the health detail in the log.

File: qa/workunits/cephtool/test.sh (hunk 8 above)
Comment (dzafman) [inline qa/workunits/cephtool/test.sh]: yes

**Comment by dzafman** [comment]
@dachary Yes, I'm moving get_admin_socket() to test.sh, but I ran the test and in the rados suite case I don't have the correct path to the socket.  Ugh!  I don't know what the path is.   "/var/run/ceph/ceph-$client.asok" is WRONG.

**Comment by dzafman** [comment]
@dachary I figured out the asok path and even got the cluster name out of CEPH_ARGS for the qa suite run.

**Comment by dzafman** [comment]
PASSED TEST:  http://pulpito.ceph.com/dzafman-2017-04-20_15:36:26-rados-wip-zafman-testing-distro-basic-smithi/

**Pull request merged**